mlflow/models/utils.py

```diff
@@ -171,23 +171,28 @@ def _handle_dataframe_input(input_ex):
                     "- bytes\n"
                     "but got '{}'".format(type(input_example)),
```

Comment:
Just a nit for keeping this code readable - seems a little odd to throw here now saying that we expected a numpy array when passing a numpy array to this `_coerce_to_pandas_df` method would technically be invalid (right)? I wonder if we should move this exception up into the `__init__` method, e.g. if the input wasn't a sparse matrix or ndarray and we couldn't cast the input into a pandas DF, throw there with this error message. We could achieve that by having this method return `None` if the input is non-tabular and checking for that in `__init__`